packages/@aws-cdk/cdk/lib/context.ts

```diff
@@ -16,95 +22,108 @@ const SSM_PARAMETER_PROVIDER = 'ssm';
       return defaultValue;
     }
-    const key = colonQuote([provider].concat(scope).concat(args)).join(':');
-    const value = this.context.getContext(key);
+
+    const value = this.context.getContext(this.key);
+
     if (value != null) {
-      if (typeof value !== 'string') {
-        throw new TypeError(`Expected context parameter '${key}' to be a string, but got '${value}'`);
-      }
       return value;
     }
 
-    this.stack.reportMissingContext(key, { provider, scope, args });
+    this.stack.reportMissingContext(this.key, {
+      provider: this.provider,
+      props: this.props,
+    });
     return defaultValue;
   }
-
   /**
-   * Read a provider value, verifying it's a list
-   * @param provider The name of the context provider
-   * @param scope The scope (e.g. account/region) for the value
-   * @param args Any arguments
+   * Read a provider value, verifying it's a string
    * @param defaultValue The value to return if there is no value defined for this context key
    */
-  public getStringListValue(
-    provider: string,
-    scope: undefined | string[],
-    args: string[],
-    defaultValue: string[]): string[] {
+  public getStringValue( defaultValue: string): string {
     // if scope is undefined, this is probably a test mode, so we just
-    // return the default value and report an error so this in not accidentally used
-    // in the toolkit
-    if (!scope) {
-      // tslint:disable-next-line:max-line-length
-      this.context.addError(formatMissingScopeError(provider, args));
+    // return the default value
+    if (!this.props.account || !this.props.region) {
+      this.context.addError(formatMissingScopeError(this.provider, this.props));
       return defaultValue;
     }
 
-    const key = colonQuote([provider].concat(scope).concat(args)).join(':');
-    const value = this.context.getContext(key);
+    const value = this.context.getContext(this.key);
 
     if (value != null) {
-      if (!value.map) {
-        throw new Error(`Context value '${key}' is supposed to be a list, got '${value}'`);
+      if (typeof value !== 'string') {
+        throw new TypeError(`Expected context parameter '${this.key}' to be a string, but got '${JSON.stringify(value)}'`);
       }
       return value;
     }
 
-    this.stack.reportMissingContext(key, { provider, scope, args });
+    this.stack.reportMissingContext(this.key, {
+      provider: this.provider,
+      props: this.props,
+    });
     return defaultValue;
   }
 
   /**
-   * Helper function to wrap up account and region into a scope tuple
+   * Read a provider value, verifying it's a list
+   * @param defaultValue The value to return if there is no value defined for this context key
    */
-  public accountRegionScope(providerDescription: string): undefined | string[] {
-    const stack = Stack.find(this.context);
-    if (!stack) {
-      throw new Error(`${providerDescription}: construct must be in a stack`);
-    }
+  public getStringListValue(
```

Comment:
I think I can dry this function up by passing a function here. I don't think I can use generics here. I suppose I could extract this to a non exported function and pass everything into it, but I think just passing a `validateType(value: any): boolean{}` will probably do?

Reply:
actually that is also blocked by JSII ... I'm not sure we can really DRY this up much then?

Reply:
It's fine as-is.